flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -128,8 +128,9 @@ public RemoteInputChannel(
 	void assignExclusiveSegments() throws IOException {
 		checkState(bufferManager.unsynchronizedGetAvailableExclusiveBuffers() == 0,
 			"Bug in input channel setup logic: exclusive buffers have already been set for this input channel.");
+		checkState(initialCredit > 0, "Number of exclusive buffers must be larger than 0.");
```

Comment:
nit: if we want to check this argument, it would be better to be done in the initialization of constructor. 
I think we can remove it directly since the passed `networkBuffersPerChannel` in constructor should already be checked in previous component.